handler/file/retrieve.go

```diff
@@ -114,13 +114,86 @@ func (r *filecoinReader) Read(p []byte) (int, error) {
 		readLen = remainingBytes
 	}
 
+	buf := bytes.NewBuffer(p)
+	buf.Reset()
+
+	n, err := r.writeToN(buf, readLen)
+	return int(n), err
+}
+
+func (r *filecoinReader) WriteTo(w io.Writer) (int64, error) {
```

Comment:
```suggestion
// WriteTo is implemented in order to directly handle io.Copy / io.CopyN operations
// rather than allow small, separate Read operations.
func (r *filecoinReader) WriteTo(w io.Writer) (int64, error) {
```

or something like that, just to be clear why we're doing this

Reply:
FYI, As the reader, this does not apply to `io.CopyN` since `io.CopyN` wraps the data soruce in a `LimitReader`

Reply:
Where is this used? I don't find io.WriterTo being used as filecoinReader

Reply:
This is called within `io.Copy` if available, so that copying through the default buffer can be avoided, [here](https://cs.opensource.google/go/go/+/refs/tags/go1.21.3:src/io/io.go;l=409-413).

Reply:
> Where is this used? I don't find io.WriterTo being used as filecoinReader

This is called within `io.Copy` if available, so that copying through the default buffer can be avoided, [here](https://cs.opensource.google/go/go/+/refs/tags/go1.21.3:src/io/io.go;l=409-413).